### PR TITLE
docs: fix percentage columns demo

### PR DESCRIPTION
**Pull request by AlanCutFlim**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #29831 

### 💡 Background and solution
There is a different behaviour when `Col.flex` sets to a number or string value.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix percentage columns demo      |
| 🇨🇳 Chinese |     修复 Col 百分比示例      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/grid/demo/flex-stretch.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ ReactDOM.render(
   <>
     <Divider orientation="left">Percentage columns</Divider>
     <Row>
-      <Col flex={2}>2 / 5</Col>
-      <Col flex={3}>3 / 5</Col>
+      <Col flex="2">2 / 5</Col>
```
Comment (xrkffgg) [inline components/grid/demo/flex-stretch.md:23]: 为啥改呢？

Comment (AlanCutFlim) [reply components/grid/demo/flex-stretch.md:23]: https://codesandbox.io/s/flex-percentage-column-issue-forked-slnkv?file=/index.js
`Col.flex` 设置 string 时宽度是正确的，设置 number 宽度不对

Comment (xrkffgg) [reply components/grid/demo/flex-stretch.md:23]: 这个不止是 Demo 的问题，先确认下不同类型具体的表现行为，也可能是个 Bug

Comment (AlanCutFlim) [reply components/grid/demo/flex-stretch.md:23]: 我也好奇为什么不同类型的值要设计成有不同的行为，这个是设计如此吗？ @afc163 @zombieJ

**Comment by li-jia-nan** [comment]
冲突了，先close了抱歉，如果有需要可以重新发起

**Pull request closed without merging**